cpu/cc2538/periph/hwrng.c

```diff
@@ -21,22 +21,30 @@
  * @}
  */
 
+#include "vendor/hw_memmap.h"
+#include "vendor/hw_soc_adc.h"
+
 #include "cpu.h"
 #include "periph/hwrng.h"
 
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+static cc2538_soc_adc_t *soc_adc = (cc2538_soc_adc_t *)SOC_ADC_BASE;
+
 void hwrng_init(void)
 {
     uint16_t seed = 0;
     int i;
 
     /* Make sure the RNG is on */
-    SOC_ADC->cc2538_adc_adccon1.ADCCON1bits.RCTRL = 0;
+    soc_adc->ADCCON1 = (0UL << SOC_ADC_ADCCON1_RCTRL_S);
```

Comment:
Any reason why you are shifting 0 to 0?

Reply:
need to ensure that RCTRL is 0, I changed this to bit unset operation.